[omnibus] strip symbols

truthbk opened the pull request:
### What does this PR do?

This PR strips the symbols to the binaries/libs on linux. 

The current problem with the PR is these symbols are currently deleted after being stripped, we don't want to do this, we want to zip them up and upload them somewhere. So that's pending.  

### Motivation

Decrease package size.

### Additional Notes

Please make sure we save the symbols somewhere.

Sister PR: https://github.com/DataDog/omnibus-ruby/pull/91 (all heavy lifting done there)

truthbk committed:
[omnibus] strip symbols - linux - wip

truthbk committed:
[omnibus] splitting strip steo - previous attempt was a NOP

truthbk committed:
[omnibus][ostool] actual debug statement for symbol stripping

truthbk committed:
[gitlab] omnibus build on debug log-level for troubleshooting purposes

truthbk committed:
[gitlab][skip ci] no need for debug levels

truthbk committed:
Merge branch 'master' into jaime/strip-symbols

truthbk committed:
[strip] comment some stuff out

truthbk committed:
[omnibus] omnibus-ruby now responsible for stripping logic

truthbk committed:
[strip+debug] stripping logic in omnibus-ruby + dbg install scripts

truthbk committed:
[strip+dbg] empty dbg install scripts + trigger stripping and debug build

truthbk committed:
[omnibus] some cleanup, heavy lifting in omnibus-ruby

truthbk committed:
[omnibus] some cleanup, heavy lifting in omnibus-ruby

truthbk committed:
[omnibus] some more cleanup

truthbk committed:
[omnibus] cleaner debug build at the project level

KSerrania reviewed (changes requested):
As I mentioned in the related [omnibus-ruby PR](https://github.com/DataDog/omnibus-ruby/pull/91), the `.gitlab-ci.yml` file also needs to be updated:
- [this line](https://github.com/DataDog/datadog-agent/blob/667605285a2f65d9bd5c5281b5d13be97a6a9bdb/.gitlab-ci.yml#L354) fails in the Debian build because now two files match the expression, and the `-c` option can only take one,
- [this line](https://github.com/DataDog/datadog-agent/blob/667605285a2f65d9bd5c5281b5d13be97a6a9bdb/.gitlab-ci.yml#L417) of the RPM build fails because it tries to install the debug package, which fails because it's missing a dependency (the main package).

truthbk committed:
[release] restore omnibus-ruby branch

truthbk committed:
[gitlab] addressing multi-artifact scenarios

[gitlab] fixing pipelines

albertvaka commented on omnibus/config/projects/agent.rb:
```diff
@@ -191,3 +191,8 @@
+
+if linux?
+  strip_build true
+  debug_path ".debug"  # the srip symbols will be in here
```
You don't reference this `.debug` directory anywhere else? Just making sure this is intended.

truthbk replied:
Yes, this is intended, the `.debug` path is created by the stripper by `omnibus-ruby`. I'll add a clarifying note.

truthbk committed:
[dockerfile] build should ignore dbg deb artifact

truthbk committed:
[omnibus] don't strip psycopg2

truthbk commented:
This sister PR will also need to be merged before this one: https://github.com/DataDog/omnibus-ruby/pull/94

truthbk committed:
[omnibus] clarifying comment

truthbk committed:
[dockerfile] use egrep to succeed if we got matches.

permission denied errors can break the chain, egrep will
return 0 if we got any matches in the find, which will
allow the chain of commands to continue,

[dockerfile] extract the verbose way for proper processing

truthbk committed:
Merge branch 'master' into jaime/strip-symbols

KSerrania reviewed (approved):
LGTM, there's a problem with the package sizes of the `deb` packages, but it has to do with the 
`omnibus-ruby` PR.
